derive_stringset/src/lib.rs

```diff
@@ -70,3 +79,41 @@ pub fn derive(input: TokenStream) -> TokenStream {
 
     TokenStream::from(expanded)
 }
+
+fn derive_enum(input: &syn::DeriveInput) -> TokenStream {
+    let variants = if let syn::Data::Enum(syn::DataEnum { ref variants, .. }) = input.data {
+        variants
+    } else {
+        // We should not have called derive_enum() if input.data wasn't an enum
+        unreachable!();
+    };
+
+    // Only support simple enums, check each field
+    for v in variants.iter() {
+        match v.fields {
+            syn::Fields::Named(_) | syn::Fields::Unnamed(_) => {
+                unimplemented!("only simple enums are supported")
+            }
+            syn::Fields::Unit => {}
+        }
+    }
+
+    let name = &input.ident;
+    let expanded = quote! {
+        impl StringSet for #name {
```

Comment:
Needs to be `impl miniconf::StringSet for #name` or users may get an error indicating that `StringSet` is implemented, but not in scope. See #19